src/app/components/AddSource.svelte

```diff
@@ -0,0 +1,189 @@
+
+  const leftText = "UPLOAD";
+  const rightText = "DIRECT ENTRY";
+  export let addSource: boolean = false;
+  let manualEntry: boolean = false;
+  let onDraggedOver = false;
+  let fileList: File[] | FileList = [];
+
+  function fileFromDataTransferItem(items: DataTransferItemList): File[] {
+    const fileList: File[] = [];
+    for (let item of items) {
+      // If dropped items aren't files, reject them
+      if (item.kind === "file") {
+        const file = item.getAsFile();
+        if (file !== null) {
+          fileList.push(file);
+        }
+      }
+    }
+    return fileList;
+  }
+
+  const handleDrop = (event: DragEvent) => {
+    onDraggedOver = false;
+
+    if (event.dataTransfer == null) {
+      return;
+    } else if (event.dataTransfer.items) {
+      // Use DataTransferItemList interface to access the file(s)
+      fileList = fileFromDataTransferItem(event.dataTransfer.items);
+    } else {
+      // Use DataTransfer interface to access the file(s)
+      fileList = Array.from(event.dataTransfer.files);
+    }
+  };
+
+  const handleDragOver = () => {
+    onDraggedOver = true;
+  };
+
+  const handleDragLeave = () => {
+    onDraggedOver = false;
+  };
+
+  const handleChange = (event: Event): void => {
+    const input = event.target as HTMLInputElement;
+    if (input !== null && input.files) {
+      fileList = input.files;
+    }
+  };
+
+  const handleSave = (): void => {
+    for (let file of fileList) {
+      worker.addDictionarySourceToProject(file.name, file);
+    }
+  };
+
+  const handleClose = () => {
+    addSource = false;
+    manualEntry = false;
+  };
+
+  const handleUpload = () => {
+    addSource = true;
+    manualEntry = false;
+  };
+
+  const handleDirectEntry = () => {
+    manualEntry = true;
+  };
+</script>
+
+<style>
+  .upload-zone {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    padding: 5%;
+    min-width: 200px;
+    border: 3px dashed var(--gray-light);
+    border-radius: 5px;
+    line-height: 1.2;
+  }
+
+  .drag-over {
+    border-color: var(--blue);
+    background-color: var(--blue-light);
+  }
+
+  input {
+    display: none;
+  }
+
+  .upload-btn {
+    margin: 0 auto;
+    color: var(--primary-blue);
+  }
+
+  .upload-btn:hover {
+    cursor: pointer;
+    text-decoration: underline;
+  }
+
+  img {
+    width: 2em;
+    color: var(--blue);
+    padding-bottom: 1em;
+  }
+
+  .save-zone {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: center;
+    padding: 1%;
+  }
+
+  .arrow-button-zone {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    padding: 1%;
+  }
+</style>
+
+{#if addSource}
+  <div class="arrow-button-zone" data-cy="add-source-options">
+    <img
+      src="assets/down-arrow.svg"
+      alt="Down Arrow"
+      role="presentation"
+      class="block" />
```

Comment:
Regarding accessibility, every image that is content must have a non-empty `alt=` tag. However, some images are just visual decoration, and the app would make sense if the image was removed entirely. Decorative images should have `alt="" role="presentation"`.

**You cannot have a non-empty `alt=""` attribute plus `role="presentation"`**!

So, if the image is just a **decoration**, have this:

```suggestion
    <img
      src="assets/down-arrow.svg"
      alt=""
      role="presentation"
      class="block" />
```

However, if the image has **useful content** that _should_ be read aloud to the user, then it should look like this:

```suggestion
    <img
      src="assets/down-arrow.svg"
      alt="Down arrow"
      class="block" />
```